jrt/src/com/yahoo/jrt/MaybeTlsCryptoEngine.java

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.jrt;
+
+import java.nio.channels.SocketChannel;
+
+/**
+ * A crypto engine that supports both tls encrypted connections and
+ * unencrypted connections. The use of tls for incoming connections is
+ * auto-detected using clever heuristics. The use of tls for outgoing
+ * connections is controlled by the useTls flag given to the
+ * constructor.
+ **/
+public class MaybeTlsCryptoEngine implements CryptoEngine {
+
+    private final TlsCryptoEngine tlsEngine;
+    private final boolean useTls;
+
+    public MaybeTlsCryptoEngine(TlsCryptoEngine tlsEngine, boolean useTls) {
```

Comment:
Consider renaming `useTls` to a name that indicates that the parameter affects the client side only, e.g. `useTlsForClients`.

Reply:
will do